chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,317 @@
+
+    def puzzle_for_pk(self, pubkey: G1Element) -> Program:
+        raise NotImplementedError("vault wallet")
+
+    async def puzzle_for_puzzle_hash(self, puzzle_hash: bytes32) -> Program:
+        raise NotImplementedError("vault wallet")
+
+    async def sign_message(self, message: str, puzzle_hash: bytes32, mode: SigningMode) -> Tuple[G1Element, G2Element]:
+        raise NotImplementedError("vault wallet")
+
+    async def get_puzzle_hash(self, new: bool) -> bytes32:
+        if new:
+            return await self.get_new_puzzlehash()
+        else:
+            record: Optional[
+                DerivationRecord
+            ] = await self.wallet_state_manager.get_current_derivation_record_for_wallet(self.id())
+            if record is None:
+                return await self.get_new_puzzlehash()
+            return record.puzzle_hash
+
+    async def apply_signatures(
+        self, spends: List[Spend], signing_responses: List[SigningResponse]
+    ) -> SignedTransaction:
+        raise NotImplementedError("vault wallet")
+
+    async def execute_signing_instructions(
+        self, signing_instructions: SigningInstructions, partial_allowed: bool = False
+    ) -> List[SigningResponse]:
+        raise NotImplementedError("vault wallet")
+
+    async def path_hint_for_pubkey(self, pk: bytes) -> Optional[PathHint]:
+        raise NotImplementedError("vault wallet")
+
+    async def sum_hint_for_pubkey(self, pk: bytes) -> Optional[SumHint]:
+        raise NotImplementedError("vault wallet")
+
+    def make_solution(
+        self,
+        primaries: List[Payment],
+        conditions: Tuple[Condition, ...] = tuple(),
+        fee: uint64 = uint64(0),
+    ) -> Program:
+        raise NotImplementedError("vault wallet")
+
+    async def get_puzzle(self, new: bool) -> Program:
+        if new:
+            return await self.get_new_puzzle()
+        else:
+            record: Optional[
+                DerivationRecord
+            ] = await self.wallet_state_manager.get_current_derivation_record_for_wallet(self.id())
+            if record is None:
+                return await self.get_new_puzzle()
+            puzzle = construct_p2_delegated_secp(record.pubkey, DEFAULT_CONSTANTS.GENESIS_CHALLENGE, record.puzzle_hash)
+            return puzzle
+
+    def puzzle_hash_for_pk(self, pubkey: G1Element) -> bytes32:
+        raise ValueError("This won't work")
+
+    def require_derivation_paths(self) -> bool:
+        if getattr(self, "vault_info", None):
+            return True
+        return False
+
+    async def match_hinted_coin(self, coin: Coin, hint: bytes32) -> bool:
+        raise NotImplementedError("vault wallet")
+
+    def handle_own_derivation(self) -> bool:
+        return True
+
+    def get_recovery_info(self) -> Tuple[Optional[G1Element], Optional[uint64]]:
+        if self.vault_info.is_recoverable:
+            return self.recovery_info.bls_pk, self.recovery_info.timelock
+        return None, None
```

Comment:
Are these two things ever in the state where one is option and one isn't?  If not, I think maybe make this `Optional[Tuple[G1Element, uint64]]` instead.